doc/design/if_else_op.md

```diff
@@ -0,0 +1,55 @@
+In an if_op, only inputs with condition satisfied will be run. The op could have multiple inputs and multiple outputs.
+We should have the following design:
+
+```python
+# A 1-d bool vector
+cond = Var()
+# create an op
+if = pd.if_op()
+
+with if.true_block() as block:
+  x1 = if.input(x1)
+  x2 = if.input(x2)
+  y = pd.add(x1, x2)
+  y2 = pd.fc(x1) # contains (w,b)
+  if.output(y)
+  if.output(y2)
+  
+o1, o2 = if(cond)
+```
+
+In an if_op, only inputs with condition satisfied will be run.
+We should have the following design:
+```python
```

Comment:
IfOp should have only one branch.  An IfOp operator takes a `cond` variable whose value must be a vector of N boolean elements. Its return value has M (M<=N) instances, each corresponds to a true element in `cond`.

```python
import paddle as pd

x = var()
y = var()
cond = var()

b = pd.create_ifop_builder(inputs=[x], output_num=1)
with b.true_block():
    x = b.inputs(0)
    z = operator.add(x, y)
    b.set_output(0, operator.softmax(z))

out = b(cond)
```

If we want the output still has N instances, we can use IfElseOp with a default value, whose minibatch size must be N:

```python
import paddle as pd

x = var()
y = var()
cond = var()
default_value = var()
b = pd.create_ifelseop_builder(inputs=[x], output_num=1, default_value)
with b.true_block():
    x = b.inputs(0)
    z = operator.add(x, y)
    b.set_output(0, operator.softmax(z))

out = b(cond)
```